### PR TITLE
Flow360 2.0 documentation for volume_models, solver_numerics and outputs modules

### PR DESCRIPTION
**Pull request by angranl-flex**
The detailed added contents are listed in this PR:
https://github.com/flexcompute-readthedocs/Flow360Documentation/pull/276

### CODE REVIEW

File: flow360/component/simulation/models/solver_numerics.py [hunk 1]
```diff
@@ -149,22 +170,52 @@ class NavierStokesSolver(GenericSolverSettings):
...
-    absolute_tolerance: PositiveFloat = pd.Field(1.0e-10)
+    absolute_tolerance: PositiveFloat = pd.Field(
+        1.0e-10,
+        description="Tolerance for the NS residual, below which the solver goes to the next physical step",
+    )
 
-    CFL_multiplier: PositiveFloat = pd.Field(1.0)
-    kappa_MUSCL: pd.confloat(ge=-1, le=1) = pd.Field(-1)
+    CFL_multiplier: PositiveFloat = pd.Field(
+        1.0, description="Factor to the CFL definitions defined in :code:`Time stepping` section"
```
Comment (benflexcompute) [inline flow360/component/simulation/models/solver_numerics.py:179]: :code:`Time stepping` does not seem to work? 
image
Also can you delete the 
```
Parameters
----------
```
part? It is causing duplication:

image

File: flow360/component/simulation/models/solver_numerics.py [hunk 2]
```diff
@@ -328,9 +421,6 @@ class HeatEquationSolver(GenericSolverSettings):
         Frequency at which to solve the heat equation in conjugate heat transfer simulations
```
Comment (benflexcompute) [inline flow360/component/simulation/models/solver_numerics.py:423]: Same here. Can you remove all the 
```
Paramters
---------
```

File: flow360/component/simulation/models/solver_numerics.py [hunk 3]
```diff
@@ -379,19 +480,45 @@ class TransitionModelSolver(GenericSolverSettings):
     type_name: Literal["AmplificationFactorTransport"] = pd.Field(
         "AmplificationFactorTransport", frozen=True
     )
-    CFL_multiplier: PositiveFloat = pd.Field(2.0)
-    absolute_tolerance: PositiveFloat = pd.Field(1e-7)
-    equation_evaluation_frequency: PositiveInt = pd.Field(4)
-    turbulence_intensity_percent: Optional[pd.confloat(ge=0.03, le=2.5)] = pd.Field(None)
+    CFL_multiplier: PositiveFloat = pd.Field(
+        2.0, description="Factor to the CFL definitions defined " + 'in "Time stepping" section'
```
Comment (benflexcompute) [inline flow360/component/simulation/models/solver_numerics.py:484]:
```suggestion
        2.0, description='Factor to the CFL definitions defined in "Time stepping" section'
```

File: flow360/component/simulation/models/volume_models.py [hunk 4]
```diff
@@ -71,27 +71,39 @@ class FromUserDefinedDynamics(Flow360BaseModel):
 
 
 class ExpressionInitialConditionBase(Flow360BaseModel):
-    """:class:`ExpressionInitialCondition` class"""
+    """:class:`ExpressionInitialCondition` class for specifying the intial conditions of
+    :obj:`~flow360.component.simulation.models.solver_numerics.NavierStokesSolver`."""
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:75]: use class to be consistent throughout.

File: flow360/component/simulation/models/volume_models.py [hunk 5]
```diff
@@ -194,65 +231,148 @@ def validate_consistent_array_length(cls, values):
...
-    """Same as Flow360Param ActuatorDisks.
+    """:class:`ActuatorDisk` class for setting up the inputs for an Actuator Disk.
+    Please refer to the :ref:`actuator disk knowledge base <knowledge_base_actuatorDisks>` for further information.
     Note that `center`, `axis_thrust`, `thickness` can be acquired from `entity` so they are not required anymore.
     """
 
-    entities: EntityList[Cylinder] = pd.Field(alias="volumes")
-    force_per_area: ForcePerArea = pd.Field()
-    name: Optional[str] = pd.Field(None)
+    entities: Optional[EntityList[Cylinder]] = pd.Field(
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:239]: Can you change entities definition here to be consistent with the latest one?

File: flow360/component/simulation/models/volume_models.py [hunk 6]
```diff
@@ -194,65 +231,148 @@ def validate_consistent_array_length(cls, values):
...
+    )
+    number_of_blades: pd.StrictInt = pd.Field(gt=0, le=10, description="Number of blades to model")
+    omega: AngularVelocityType.NonNegative = pd.Field(
+        description="Nondimensional rotating speed, radians/nondim-unit-time, "
+        + "= :math:`\\frac{\\Omega*L_{gridUnit}}{C_\\infty}`"
+    )
+    chord_ref: LengthType.Positive = pd.Field(
+        description="Nondimensional reference chord used to compute sectional blade loadings"
+    )
+    n_loading_nodes: pd.StrictInt = pd.Field(
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:327]: We probabaly need to change how we specify the range:
image
VS
image

File: flow360/component/simulation/models/volume_models.py [hunk 7]
```diff
@@ -194,65 +231,148 @@ def validate_consistent_array_length(cls, values):
...
+        description="A list of :class:`BETDiskTwist` objects specifying the twist in degrees as a "
+        + "function of radial location."
+    )
+    chords: List[BETDiskChord] = pd.Field(
+        description="A list of :class:`BETDiskChord` objects specifying the blade chord as a function "
+        + "of the radial location. "
+    )
+    sectional_polars: List[BETDiskSectionalPolar] = pd.Field(
+        description="A list of :class:`BETDiskSectionalPolar` objects for every radial location specified in "
+        + ":code:`sectional_radiuses`."
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:370]: Use class? If it works

File: flow360/component/simulation/models/volume_models.py [hunk 8]
```diff
@@ -295,20 +415,29 @@ def check_bet_disk_3d_coefficients_in_polars(self):
...
+        description="The entity list for the Rotation model. "
+        + "The entity should be :class:`Cylinder` or :class:`GenericVolume` type.",
+    )
 
     # TODO: Add test for each of the spec specification.
     spec: Union[AngleExpression, FromUserDefinedDynamics, AngularVelocity] = pd.Field(
-        discriminator="type_name"
+        discriminator="type_name",
+        description="The angular velocity for rotation. Should use the unit system "
+        + "to convert degree or degree per second to radian and radian per second.",
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:436]: hmmm. It is now dimensioned so no point of manual conversion

File: flow360/component/simulation/outputs/outputs.py [hunk 9]
```diff
@@ -97,10 +108,13 @@ class TimeAverageSurfaceOutput(SurfaceOutput):
 
 
 class VolumeOutput(_AnimationAndFileFormatSettings):
-    """Volume output settings."""
+    """Volume output settings. (test the code)"""
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:111]: What is (test the code)

Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py:111]: Removed

File: flow360/component/simulation/models/volume_models.py (hunk 7 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/volume_models.py:370]: use :attr: for now.

File: flow360/component/simulation/models/volume_models.py (hunk 6 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/volume_models.py:327]: We have tested using Annotated but it does not work. We will improve this in the future.

File: flow360/component/simulation/models/volume_models.py (hunk 5 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/volume_models.py:239]: All updated

File: flow360/component/simulation/models/volume_models.py (hunk 4 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/volume_models.py:75]: Replaced :obj: with :class:

File: flow360/component/simulation/models/solver_numerics.py (hunk 3 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/solver_numerics.py:484]: Updated

File: flow360/component/simulation/models/solver_numerics.py (hunk 2 above)
Comment (angranl-flex) [reply flow360/component/simulation/models/solver_numerics.py:423]: Removed all parameters sections in the three modules documented in this PR.

**Review by benflexcompute — APPROVED** [review]

**Pull request merged**